### PR TITLE
bazel: build the //lib folder 

### LINKED ISSUE #46832: Build lib/* with Bazel

### PR DESCRIPTION
**Pull request by jhchabran**
This PR adds a bunch of fixes to be able to build `//lib/...`. 

@jbedard if you stumble across this PR, and to follow up with our conversation from yesterday, I'm adding things in separate jobs here only as a temporary measure, so I can track that I'm not breaking things as I progress through the migration. This is obviously not the final approach, which is to simply `build //...` 

The notable fix that this PR includes is to add some build directives to the `sourcegraph/scip` `go_repository` rule, in order to deal with the fact that we're using a `replace` directive in our root `go.mod` to work around the circular depedency (sourcegraph/scip depends on sourcegraph/sourcegraph/lib which then also depends on sourcegraph/scip).

@bobheadxi I had to update `internal/cmd/resources-report` along the way. There are no unit tests, so apart for building it, so I couldn't really QA it further. 

Fixes #46832 

## Test plan

I ran a main-dry-run over https://buildkite.com/sourcegraph/sourcegraph/builds/195371 to ensure merging this does not break anything.

### CODE REVIEW

File: deps.bzl [hunk 1]
```diff
@@ -6442,8 +6555,17 @@ def go_dependencies():
...
     go_repository(
         name = "com_github_sourcegraph_scip",
+        # This fixes the build for sourcegraph/scip which depends on sourcegraph/sourcegraph/lib but
+        # gazelle doesn't know how to resolve those packages from within sourcegraph/scip.
+        build_directives = [
+            "gazelle:resolve go github.com/sourcegraph/sourcegraph/lib/errors @//lib/errors",
+            "gazelle:resolve go github.com/sourcegraph/sourcegraph/lib/codeintel/lsif/protocol @//lib/codeintel/lsif/protocol",
+            "gazelle:resolve go github.com/sourcegraph/sourcegraph/lib/codeintel/lsif/protocol/reader @//lib/codeintel/lsif/protocol/reader",
+            "gazelle:resolve go github.com/sourcegraph/sourcegraph/lib/codeintel/lsif/protocol/writer @//lib/codeintel/lsif/protocol/writer",
+        ],
```
Comment (jhchabran) [inline deps.bzl:6568]: This effectively does the same thing as having the replace directive.

File: internal/cmd/resources-report/aws.go [hunk 2]
```diff
@@ -146,8 +146,9 @@ func collectAWSResources(ctx context.Context, since time.Time, verbose bool, tag
 	wait := &sync.WaitGroup{}
 
 	// iterate over regions based on accessible EC2 regions
+	allRegion := true
```
Comment (bobheadxi) [inline internal/cmd/resources-report/aws.go:149]: @jhchabran we should remove this tool, it was one of my intern projects and I do not think it was ever really that important, there are better ways to report on unused resources out there

Comment (bobheadxi) [reply internal/cmd/resources-report/aws.go:149]: https://github.com/sourcegraph/sourcegraph/pull/46941

Comment (jhchabran) [reply internal/cmd/resources-report/aws.go:149]: Perfect :)

File: .bazelignore [hunk 3]
```diff
@@ -26,3 +26,5 @@ client/plugin-backstage/node_modules
 
 cmd/symbols/squirrel/test_repos/starlark
 docker-images/syntax-highlighter/crates/sg-syntax/src/snapshots/
+
+lib/codeintel/precise/diff/testdata/
```
[the remark this answers is not in the document]
Comment (keegancsmith) [reply .bazelignore:30]: can you use a glob? Likely this isn't the only place we have go code that lives under testdata dir. Go compiler always ignores it.

[the remark this answers is not in the document]
Comment (jhchabran) [reply .bazelignore:30]: @keegancsmith we cannot sadly, see https://github.com/bazelbuild/bazel/issues/7093

**Review by willdollman — APPROVED** [review]

**Pull request merged**